packages/sales-api-service/src/services/recurring-payments.service.js

```diff
@@ -0,0 +1,55 @@
+import { Contact, findByDateRange, findByExample, findById, permissionForLicensee, RecurringPayment } from '@defra-fish/dynamics-lib'
+
+export const getRecurringPayments = async date => {
+  // create rp thats active with no cancelled date
+  const recurringPayment = new RecurringPayment()
+  recurringPayment.cancelledDate = null
+  // grab all rp matching no cancelled date and status active
+  const activeRecurringPayments = await findByExample(recurringPayment)
+  // grab all rp matching above + within date range of current date -2,-4,-6,-8,-10
+  const dueRecurringPayments = await findByDateRange(activeRecurringPayments, date)
```

Comment:
See my above comments on the filtering. I think you need to have a method to retrieve the uncancelled, active recurring payments for the specified date